tests/xharness/Harness.cs

```diff
@@ -443,14 +445,25 @@ void ConfigureMac ()
  
 			foreach (var proj in MacTestProjects.Where ((v) => !v.GenerateVariations)) {
 				var file = proj.Path;
-				var unifiedMobile = new MacUnifiedTarget (true, false, true)
-				{
- 					TemplateProjectPath = file,
- 					Harness = this,
-					IsNUnitProject = proj.IsNUnitProject,
- 				};
-				unifiedMobile.Execute ();
-				hardcoded_unified_targets.Add (unifiedMobile);
+
+				MacUnifiedTarget unifiedTarget = null;
+				if (proj.GenerateModern) {
+					unifiedTarget = new MacUnifiedTarget (true, false, true, true) {
+						TemplateProjectPath = file,
+						Harness = this,
+						IsNUnitProject = proj.IsNUnitProject,
+					};
+				}
+				else if (proj.GenerateFull) {
+					unifiedTarget = new MacUnifiedTarget (false, false, true, true) {
+						TemplateProjectPath = file,
+						Harness = this,
+						IsNUnitProject = proj.IsNUnitProject,
+					};
+				}
```

Comment:
It looks like the target creations are identical except for a single parameter, so this could be replaced with:

```csharp
MacUnifiedTarget = new MagUnifiedTarget (project.GenerateModern, false, true, true) {
    TemplateProject = file,
    Harness = this,
    IsNUnitProject = project.IsNUnitPorject,
};
unifiedTarget.Execute ();
hardcoded_unified_targets.Add (unifiedTarget);
```

Reply:
That's silly of me. I have no clue why I didn't see that. Fixing.